File: poker-webapp/src/main/java/com/sap/ase/poker/model/Card.java
package com.sap.ase.poker.model;

public class Card implements Comparable<Card> {

	public enum Suit {
		DIAMONDS("diamonds"), HEARTS("hearts"), SPADES("spades"), CLUBS("clubs");

		public final String value;

		private Suit(String value) {
			this.value = value;
		}
	}

	public enum Kind {
		TWO(1, "2"), THREE(2, "3"), FOUR(3, "4"), FIVE(4, "5"), SIX(5, "6"), SEVEN(6, "7"), EIGHT(7, "8"), NINE(8,
				"9"), TEN(9, "10"), JACK(10, "Jack"), QUEEN(11, "Queen"), KING(12, "King"), ACE(13, "Ace");

		public final int rank;
		public final String value;

		private Kind(int rank, String value) {
			this.rank = rank;
			this.value = value;
		}
	}

	private Card.Kind kind;
	private Card.Suit suit;

	public Card(Kind kind, Suit suit) {
		this.kind = kind;
		this.suit = suit;
	}

	public Kind getKind() {
		return kind;
	}

	public Suit getSuit() {
		return suit;
	}

	@Override
	public int compareTo(Card c) {
		return c.getKind().rank - this.kind.rank;
	}
}

File: src/main/webapp/controls/Card.js
sap.ui.define([ "sap/ui/core/Control" ], function(Control) {
	"use strict";

	// Unicodes: https://en.wikipedia.org/wiki/Playing_cards_in_Unicode

	function suitToUnicodeChar(suit) {
		if (suit === "spades") {
			return "A";
		} else if (suit === "hearts") {
			return "B";
		} else if (suit === "diamonds") {
			return "C";
		} else if (suit === "clubs") {
			return "D";
		} else {
			throw new Error("unknown suit: " + suit);
		}
	}

	function kindToUnicodeChar(kind) {
		if (kind === "ace") {
			return "1";
		} else if (parseInt(kind) >= 2 && parseInt(kind) <= 9) {
			return kind;
		} else if (kind === "10") {
			return "A";
		} else if (kind === "jack") {
			return "B";
		} else if (kind === "queen") {
			return "D";
		} else if (kind === "king") {
			return "E";
		} else {
			throw new Error("unknown kind: " + kind);
		}
	}
	
	var toUnicode = function(suit, kind) {
		return "&#x1F0" + suitToUnicodeChar(suit) + kindToUnicodeChar(kind);
	};

	var getFontColor = function(suit) {
		return (suit === "hearts" || suit === "diamonds") ? "red" : "black";
	}

	return Control.extend("poker.controls.Card", {
		metadata : {
			properties : {
				suit : {
					type : "string"
				},
				kind : {
					type : "string"
				}
			}
		},
		renderer : function(oRM, oControl) {
			oRM.write("<div");
			oRM.writeControlData(oControl);
			oRM.writeClasses();
			if (getFontColor(oControl.getSuit()) === "red") {
				oRM.write("><div style=\"font-size:5.5em;color:red;\">");
			} else {
				oRM.write("><div style=\"font-size:5.5em;color:black;\">");
			}
			oRM.write(toUnicode(oControl.getSuit(), oControl.getKind()));
			oRM.write("</div></div>");
		}
	});
});